openvdb/points/PointMask.h

```diff
@@ -0,0 +1,384 @@
+///        and while evaluating the point filter
+template <typename GridT, typename PointDataGridT, typename FilterT>
+struct PointsToTransformedScalarOp
+{
+    using LeafManagerT = tree::LeafManager<const typename PointDataGridT::TreeType>;
+    using ValueType = typename GridT::TreeType::LeafNodeType::ValueType;
+    using HandleT = AttributeHandle<Vec3f>;
+    using CombinableT = typename GridCombinerOp<GridT>::CombinableT;
+
+    PointsToTransformedScalarOp(const math::Transform& newTransform,
+                         const math::Transform& transform,
+                         const FilterT& filter,
+                         CombinableT& combinable)
+        : mNewTransform(newTransform)
+        , mTransform(transform)
+        , mFilter(filter)
+        , mCombinable(combinable) {}
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const
+    {
+        bool exists;
+        auto& grid = mCombinable.local(exists);
+        if (!exists) {
+            grid.setTransform(mNewTransform.copy());
+        }
+
+        auto& countTree = grid.tree();
+
+        for (auto leaf = range.begin(); leaf; ++leaf) {
+
+            auto handle = HandleT::create(leaf->template constAttributeArray("P"));
+
+            for (auto iter = leaf->template beginIndexOn(mFilter); iter; iter++) {
+                const Vec3d position = mTransform.indexToWorld(handle->get(*iter) +
+                    iter.getCoord().asVec3d());
+                const Coord ijk = mNewTransform.worldToIndexCellCentered(position);
+
+                auto* countLeaf = countTree.touchLeaf(ijk);
+                countLeaf->modifyValue(ijk, tools::valxform::SumOp<ValueType>(1));
+            }
+        }
+    }
+
+private:
+    const openvdb::math::Transform& mNewTransform;
+    const openvdb::math::Transform& mTransform;
+    const FilterT& mFilter;
+    CombinableT& mCombinable;
+}; // struct PointsToTransformedScalarOp
+
+
+template<typename PointDataGridT, typename GridT>
+inline typename GridT::Ptr convertPointsToScalar(
+    const PointDataGridT& points,
+    const std::vector<Name>& includeGroups,
+    const std::vector<Name>& excludeGroups)
+{
+    using point_mask_internal::PointsToScalarOp;
+
+    using GridTreeT = typename GridT::TreeType;
+    using ValueT = typename GridTreeT::ValueType;
+
+    // copy the topology from the points grid
+
+    typename GridTreeT::Ptr tree(new GridTreeT(points.constTree(),
+        false, openvdb::TopologyCopy()));
+    typename GridT::Ptr grid = GridT::create(tree);
+    grid->setTransform(points.transform().copy());
+
+    // early exit if no leaves
+
+    if (points.constTree().leafCount() == 0)            return grid;
+
+    const bool useGroup = includeGroups.size() > 0 || excludeGroups.size() > 0;
+
+    // early exit if mask and no group logic
+
+    if (std::is_same<ValueT, bool>::value && !useGroup) return grid;
+
+    tree->voxelizeActiveTiles();
```

Comment:
this shouldn't be necessary as point data grids shouldn't contain active tiles? If it does then PointsToScalarOp will also break as it attempts to probe leaf nodes

Reply:
I'll remove it. Certainly a rare edge case, but I'm not sure there's anything preventing the use of active tiles with point data grids.